File: Go/Subtract-The-Product-And-Sum-Of-Digits-Of-An-Integer.go
package main

import "strconv"

func main() {

}

func subtractProductAndSum(n int) int {
	sum, product := 0, 1

	for _, digit := range strconv.Itoa(n) {
		val, _ := strconv.Atoi(string(digit))
		sum += val
		product *= val
	}

	return product -sum
}

File: Rust/Subtract-The-Product-And-Sum-Of-Digits-Of-An-Integer.rs
pub struct Solution {}

impl Solution {
    pub fn subtract_product_and_sum(n: i32) -> i32 {
        let n = n.to_string();
        let (mut sum, mut product) = (0, 1);
        for number in n.chars() {
            let num = number.to_digit(10).unwrap();
            sum += num;
            product *= num;
        }

        (product - sum) as i32
    }
}
fn main() {}
